src/realm/object-store/sync/app.cpp

```diff
@@ -777,6 +778,51 @@ void App::remove_user(std::shared_ptr<SyncUser> user, std::function<void(Optiona
     }
 }
 
+void App::delete_user(std::shared_ptr<SyncUser> user, util::FunctionRef<void(Optional<AppError>)> completion_block)
+{
+    if (!user || user->state() == SyncUser::State::Removed) {
+        return completion_block(
+            AppError(make_client_error_code(ClientErrorCode::user_not_found), "User has already been removed"));
+    }
+
+    auto users = m_sync_manager->all_users();
+
+    auto it = std::find(users.begin(), users.end(), user);
+
+    if (it == users.end()) {
+        return completion_block(
+            AppError(make_client_error_code(ClientErrorCode::user_not_found), "No user has been found"));
+    }
+
+    std::string route = util::format("%1/auth/session", m_base_route);
+
+    Request req;
+    req.method = HttpMethod::del;
+    req.url = route;
+    req.timeout_ms = m_request_timeout_ms;
+    req.uses_refresh_token = true;
+    req.headers = get_request_headers();
+    req.headers.insert({"Authorization", util::format("Bearer %1", user->refresh_token())});
+    {
+        std::lock_guard<std::mutex> lock(*m_route_mutex);
+        req.url = util::format("%1/auth/delete", m_base_route);
+    }
+
+    do_request(req, [anchor = shared_from_this(), completion_block = std::move(completion_block), this,
+                     &user](Response response) {
```

Comment:
Not a huge deal, but it looks like we only need to capture a string which is a copy of `user->identity()` instead of keeping around the whole user.